### ISSUE TITLE
Add cron

### ISSUE THREAD

**Issue by elplatt** [elplatt/seltzer#316]
Paul B from Dallas Makerspace says:

> I was wondering if you guys would want to integrate the access controller we're
> using - with seltzer. I'll send you a 4-door controller for free. You might need to
> code an optional "hook" + cronjob into seltzer so it can call a webservice
> when unpaid people are "deactivated" after a grace period.

[Drupal's approach](https://api.drupal.org/api/drupal/modules%21system%21system.api.php/function/hook_cron/7) would probably work quite well.  We add a cron.php file that loads the core and runs hook_cron() for every module.  Then all the user has to do is add "wget http://localhost/cron.php" to their crontab.

Approval?  Objections?

**Comment by ramgarden** [comment]
I would also like to add a few web service REST endpoints to Seltzer.  Any good places to stick those in already?  I was imagining something like http://yourserver.com/crm/GetKeyWhiteList.php?allinfo=true&f=JSON that would be called from the door lock computer each day at 1 am or something to update its local whitelist.
That would do the queries required to find a list of all the members who have not lapsed on payments then grab all of their key serial numbers and return it as a JSON string (f = format could also be XML,CSV or similar) that could then be parsed and saved on the door lock computer as a whitelist.  Then when people scan their RFID it could check it against the whitelist and throw the door lock or not.

**Comment by elplatt** [comment]
Yes! REST endpoints are something I've been thinking a lot about. I'm
currently prototyping a REST API that could bridge the gap between Seltzer,
future development, and other systems.

On Fri, Feb 6, 2015 at 11:32 AM, Josh P <notifications@github.com> wrote:

> I would also like to add a few web service REST endpoints to Seltzer. Any
> good places to stick those in already? I was imagining something like
> http://yourserver.com/crm/GetKeyWhiteList.php?allinfo=true&f=JSON that
> would be called from the door lock computer each day at 1 am or something
> to update its local whitelist.
> That would do the queries required to find a list of all the members who
> have not lapsed on payments then grab all of their key serial numbers and
> return it as a JSON string (f = format could also be XML,CSV or similar)
> that could then be parsed and saved on the door lock computer as a
> whitelist. Then when people scan their RFID it could check it against the
> whitelist and throw the door lock or not.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elplatt/seltzer/issues/316#issuecomment-73265314>.
>



-- 
Edward L. Platt
http://elplatt.com
http://civic.mit.edu/users/elplatt
http://i3detroit.com
@elplatt <http://twitter.com/elplatt>

This electronic mail message was sent from my desktop personal computer.
Please forgive any long-winded, overly-prosaic ramblings.

**Comment by ramgarden** [comment]
Should we create a new thread in the issues tracker for REST endpoints /
features instead of putting all the discussion here on the cron
thread/issue?

**Comment by elplatt** [comment]
Sure. I think a broader discussion needs to happen first, but it doesn't
hurt to create a stub. This conversation is just getting started on the
cooperative software mailing group:
https://groups.google.com/forum/#!forum/coop-soft

On Fri, Feb 6, 2015 at 1:37 PM, Josh P <notifications@github.com> wrote:

> Should we create a new thread in the issues tracker for REST endpoints /
> features instead of putting all the discussion here on the cron
> thread/issue?
>
> On Fri, Feb 6, 2015 at 1:09 PM, Edward L Platt <notifications@github.com>
> wrote:
>
> > Yes! REST endpoints are something I've been thinking a lot about. I'm
> > currently prototyping a REST API that could bridge the gap between
> Seltzer,
> > future development, and other systems.
> >
> > On Fri, Feb 6, 2015 at 11:32 AM, Josh P <notifications@github.com>
> wrote:
> >
> > > I would also like to add a few web service REST endpoints to Seltzer.
> Any
> > > good places to stick those in already? I was imagining something like
> > > http://yourserver.com/crm/GetKeyWhiteList.php?allinfo=true&f=JSON that
> > > would be called from the door lock computer each day at 1 am or
> something
> > > to update its local whitelist.
> > > That would do the queries required to find a list of all the members
> who
> > > have not lapsed on payments then grab all of their key serial numbers
> and
> > > return it as a JSON string (f = format could also be XML,CSV or
> similar)
> > > that could then be parsed and saved on the door lock computer as a
> > > whitelist. Then when people scan their RFID it could check it against
> the
> > > whitelist and throw the door lock or not.
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <https://github.com/elplatt/seltzer/issues/316#issuecomment-73265314>.
> > >
> >
> >
> >
> > --
> > Edward L. Platt
> > http://elplatt.com
> > http://civic.mit.edu/users/elplatt
> > http://i3detroit.com
> > @elplatt <http://twitter.com/elplatt>
> >
> > This electronic mail message was sent from my desktop personal computer.
> > Please forgive any long-winded, overly-prosaic ramblings.
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/elplatt/seltzer/issues/316#issuecomment-73284640>.
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elplatt/seltzer/issues/316#issuecomment-73289029>.
>



-- 
Edward L. Platt
http://elplatt.com
http://civic.mit.edu/users/elplatt
http://i3detroit.com
@elplatt <http://twitter.com/elplatt>

This electronic mail message was sent from my desktop personal computer.
Please forgive any long-winded, overly-prosaic ramblings.
